src/lib/Libifl/int_status.c

```diff
@@ -81,7 +221,472 @@ PBSD_status(int c, int function, char *objid, struct attrl *attrib, char *extend
 	}
 
 	/* get the status reply */
-	return (PBSD_status_get(c));
+	return (PBSD_status_get(c, NULL));
+}
+
+/**
+ * @brief
+ * aggr_job_ct:
+ *	aggregate the job counts attribute of servers
+ *	total_jobs and state_counts are the attributes aggregated here.
+ * @param[in] cur - batch status to first server
+ * @param[in] next - batch status to second server
+ * 
+ * @return void
+ */
+static void
+aggr_job_ct(struct batch_status *cur, struct batch_status *nxt)
```

Comment:
IFL is not waiting for all server response to start processing the result. It just starts with two and processes the rest. It makes sense as servers might be busy processing other things and that time is getting utilized. We can also go one step further and process the ones which are readily available instead of following the order.

If we just want to cut multiple decodes and encodes of the first server, I could save them in a static variable. This will make the function not multi-threadable. But we can revisit this later if we attempt multi-threading for this functions.

Reply:
In fact, aggr_job_ct is common for both server and queue objects. This optimization does not work with queues as there can be multiple objects with the same id. Can we take it up later if we see it as a performance bottleneck?

Reply:
Your reasoning sounds good to me.
I am not worried about performance in this case because there are only a handful of servers or queues, it just looked odd that you had to encode every time only to decode it back.